SwanProjects/swanprojects/handlers.py

```diff
@@ -0,0 +1,289 @@
+        self.finish(json.dumps(payload))
+
+class StacksInfoHandler(APIHandler):
+
+    swan_projects_config = None
+
+    def initialize(self):
+        self.swan_projects_config = SwanProjects(config=self.config)
+
+    @tornado.web.authenticated
+    def get(self):
+        """
+        This endpoint is required for the project dialog, it's returning the information saved on stacks.json
+        """
+        with open(self.swan_projects_config.stacks_path) as f:
+            stacks = json.loads(f.read())
+        self.finish(json.dumps({"stacks": stacks}))
+
+class KernelSpecManagerPathHandler(APIHandler):
+    @tornado.web.authenticated
+    def post(self):
+        """
+        This endpoint is required for the project kernel spec manager, it's it is setting the path to
+        check if we are inside a project to change the kernel spec manager path
+        """
+        input_data = self.get_json_body()
+        path = input_data["path"]
+
+        self.log.info(f"KernelSpecManagerPathHandler = {input_data}")
+
+        project = get_project_path(path)
+        if self.kernel_spec_manager.set_path(path):
+            data = {"status": True, "is_project": project is not None,
+                    "msg": f"SWAN kernel spec manager set to path: {path}"}
+            self.finish(json.dumps(data))
+        else:
+            data = {"status": False, "is_project": project is not None,
+                    "msg": f"Error setting SWAN kernel spec manager to path: {path}"}
+            self.finish(json.dumps(data))
+
+class CreateProjectHandler(APIHandler):
+    @tornado.web.authenticated
+    def post(self):
+        """
+        Endpoint to create a project, receive project information such as name, stack, platform, release, user_script.
+        The project is created at $HOME/SWAN_projects/project_name and a hidden json ".swanproject" file with the information
+        project is set inside the project folder.
+        """
+        input_data = self.get_json_body()
+        self.log.info(f"creating project {input_data}")
+
+        name = input_data["name"]
+        stack = input_data["stack"]  # CMSSW/LCG
+        platform = input_data["platform"]  # SCRAM/x86_64..centos7..gccX
+        release = input_data["release"]  # CMSSW_X_Y_Z/LCG_XYZ
+        user_script = input_data["user_script"]
+
+        project_dir = os.environ["HOME"] + "/SWAN_projects/" + name
+        try:
+            os.makedirs(project_dir)
+        except Exception as msg:
+            data = {"status": False, "project_dir": f"SWAN_projects/{name}",
+                    "msg": f"Error creating folder for project {name}, traceback: {msg}"}
+            self.finish(json.dumps(data))
+            return
+        swan_project_file = project_dir + os.path.sep + '.swanproject'
+        swan_project_content = {'stack': stack, 'release': release,
+                                'platform': platform}
+        try:
+            with open(swan_project_file, 'w+') as f:
```

Comment:
You should use the contents methods to create files in user home dir. Same for creating dirs etc.
For example, we will replace swancontents with a CS3APIs compatible contents module. This will not write in the filesystem (and to EOS via FUSE), but instead it will talk to CERNBox. If you don't use those methods, this will break,

Reply:
contents manager was implemented.
fixed.